File: app/src/main/java/androidstudio/class24/java/FragmentClass24Output2.java
package androidstudio.class24.java;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.rafaelmfer.codeclasses.R;

import static androidstudio.class24.java.FragmentClass24Input1.NAME_KEY;

public class FragmentClass24Output2 extends Fragment {

    TextView tvWelcomeName;
    Button btFrag2LetsGo;

    private InterfaceFragments activity;

    public FragmentClass24Output2() {
    }

    public static FragmentClass24Output2 newInstance(Bundle bundle) {
        FragmentClass24Output2 frag = new FragmentClass24Output2();
        frag.setArguments(bundle);
        return frag;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_class24_2, container, false);
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        activity = (InterfaceFragments) context;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        bindViews(view);

        tvWelcomeName.setText(getString(R.string.welcome_name, getArguments().get(NAME_KEY)));
        btFrag2LetsGo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(view.getContext(), Class24Activity2.class));
            }
        });
    }

    private void bindViews(View view) {
        tvWelcomeName = view.findViewById(R.id.tvWelcomeName);
        btFrag2LetsGo = view.findViewById(R.id.btFrag2LetsGo);
    }
}

File: app/src/main/java/androidstudio/class24/kotlin/FragmentClass24Output2.kt
package androidstudio.class24.kotlin

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidstudio.class24.kotlin.FragmentClass24Input1.Companion.NAME_KEY
import androidx.fragment.app.Fragment
import com.rafaelmfer.codeclasses.R
import kotlinx.android.synthetic.main.fragment_class24_2.*

class FragmentClass24Output2 : Fragment() {

    private var activity: InterfaceFragments? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_class24_2, container, false)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        activity = context as InterfaceFragments
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        tvWelcomeName.text = getString(R.string.welcome_name, arguments?.get(NAME_KEY))
        btFrag2LetsGo.setOnClickListener { startActivity(Intent(it.context, Class24Act2::class.java)) }
    }

    companion object {
        fun newInstance(bundle: Bundle?): FragmentClass24Output2 {
            val frag = FragmentClass24Output2()
            frag.arguments = bundle
            return frag
        }
    }
}